name: aurum_devdocker
services:
  app_init: #we use this container to fix the permissions of the app volume to make it usable for local user
    image: node:22.3-bookworm
    volumes:
      - ../:/home/node/app
      - ./data/var/aurum/gpg:/var/aurum/gpg
    entrypoint:
      - sh
      - -c
      - |
        chown -R ${UID}:${GID} /var/aurum/gpg
  app:
    restart: always
    depends_on:
      app_init:
        condition: service_completed_successfully
      database:
        condition: service_healthy
    user: ${UID}:${GID}
    working_dir: /home/node/app
    image: node:22.3-bookworm
    ports:
      - "80:5173"
      - "9229:9229"
    env_file:
      - path: ./.database.env
        required: true
      - path: ./.app.env
        required: true
    command: npm run dev
    volumes:
      - ../:/home/node/app
      - ./data/var/aurum/gpg:/var/aurum/gpg

  database_init: #we use this container to fix the permissions of the database volume to make it usable for local user
    image: postgres:16.3-bookworm
    volumes:
      - ./data/pgsql:/var/lib/postgresql/data
    env_file:
      - path: ./.database.env
        required: true
    entrypoint:
      - sh
      - -c
      - |
        chown -R ${UID}:${GID} /var/lib/postgresql/data
  database:
    restart: always
    depends_on:
      - database_init
    user: ${UID}:${GID}
    image: postgres:16.3-bookworm
    ports:
      - "127.0.0.1:5432:5432"
    env_file:
      - path: ./.database.env
        required: true
    volumes:
        - ./data/pgsql:/var/lib/postgresql/data
    healthcheck:
        test: ["CMD-SHELL", "pg_isready -U $$POSTGRES_USER"]
        interval: 5s
        timeout: 5s
        retries: 5